### ISSUE TITLE
This might be caused by classes that are not decorated by @nni.trace. Another option is to force bytes pickling and try to raise pickle_size_limit.

### ISSUE THREAD

**Issue by ShrutiSarikaChakraborty** [microsoft/nni#5559]
**Describe the issue**:

**Environment**:
- NNI version:
- Training service (local|remote|pai|aml|etc):
- Client OS:
- Server OS (for remote mode only):
- Python version:
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?:

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:

**Comment by ShrutiSarikaChakraborty** [comment]
Hi,

This is not working for me . Since the last thread is closed, hence opening a new one. I have just started using NNI. I would be very grateful if you could please help.

Error:

File "C:\Users\ADMIN\anaconda3\lib\site-packages\nni\common\serializer.py", line 864, in _json_tricks_any_object_encode
raise PayloadTooLarge(f'Pickle too large when trying to dump {obj}. This might be caused by classes that are '

PayloadTooLarge: Pickle too large when trying to dump tensor([[[0., 0., 0., ..., 0., 0., 0.],
[0., 0., 0., ..., 0., 0., 0.],
[0., 0., 0., ..., 0., 0., 0.],
...,
[0., 0., 0., ..., 0., 0., 1.],
[0., 0., 0., ..., 0., 0., 1.],
[0., 0., 0., ..., 0., 0., 1.]]], device='cuda:0'). This might be caused by classes that are not decorated by @nni.trace. Another option is to force bytes pickling and try to raise pickle_size_limit.

What I did,

'''splitting train-test data'''

train_data,test_data, train_labels,test_labels = train_test_split(data, labels, test_size=.2)

@nni.trace
def get_train_dataset():
train_dataDataset = TensorDataset(train_data,train_labels)
return (train_dataDataset)

@nni.trace
def get_test_dataset():
test_dataDataset = TensorDataset(test_data,test_labels)
return (test_dataDataset)

create dataloader objects
train_loader = nni.trace(DataLoader)(get_train_dataset(),batch_size=15, shuffle=True, drop_last=True)
test_loader = nni.trace(DataLoader)(get_test_dataset(), batch_size=20, shuffle=True, drop_last=True)

#In evaluate model,

def evaluate_model(): #argument should be equal to search-space params
device = torch. Device('cuda') if torch.cuda.is_available() else torch.device('cpu')
model_space.to(device)
numepochs = 10
trainAcc = []
SkAcc = []
losses = []
learning_rate = 0.00018
L2lambda = 0
momentum = 0
lossfun,optimizer = OptandLoss(model_space,learning_rate,momentum,L2lambda,'Adam')
trainAcc,SkAcc,losses, BestModel = trainTheModel(model_space,numepochs,lossfun,optimizer,train_loader,test_loader,trainAcc,SkAcc,losses)
finalSkAcc = SkAcc[-1]
nni.report_final_result(finalSkAcc)
return nni.report_final_result(finalSkAcc)

Thanks for your help. Please let me know if I should send you the notebook.
Regards,
Shruti

**Comment by matluster** [comment]
Why do you need to put `train_loader` and `test_loader` outside `evaluate_model`? I suggest putting all of them into the evaluate function and try again.

Also your format looks messy. I suggest properly formatting them with markdown syntax.

**Comment by Lijiaoa** [comment]
any updates? @ShrutiSarikaChakraborty
